transducers/shared/src/main/scala/fs2/data/stt/Table.scala

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2021 Lucas Satabin
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fs2.data.stt
+
+import scala.annotation.implicitNotFound
+
+/** A typeclass indicating that some type `T` can be used as a lookup table.
+  */
+@implicitNotFound("Cannot prove that type ${T} can be used as a lookup table. Make sure to provide an implicit instance of `fs2.data.stt.Table[${T}]` in scope")
+trait Table[T[_, _]] {
+  def get[From, To](m: T[From, To])(from: From): Option[To]
+}
+
+object Table {
+
+  implicit object PartialFunctionTable extends Table[PartialFunction] {
+    def get[From, To](m: PartialFunction[From, To])(from: From): Option[To] =
+      m.lift(from)
+  }
+
+  implicit object MapTable extends Table[Map] {
```

Comment:
`Map <: MapOps <: PartialFunction`, so we might run into ambiguous implicit here. Maybe move the partial function instance into a low priority trait to be safe?

Reply:
> Not entirely sure what it does without docs, but left some small Scala-specific comments and will stay tuned for updates smile

Documentation will come. This is a first basic building block for higher-level features. More transducers to come soon, I hope, especially SSTs, which will allow for parsing. This STT can be the base for streaming query languages on tree structures.